sdk/core/azure-core/src/test/java/com/azure/core/implementation/jackson/AnimalWithTypeIdContainingDot.java

```diff
@@ -24,4 +27,54 @@
         value = RabbitWithTypeIdContainingDot.class)
 })
 public abstract class AnimalWithTypeIdContainingDot implements JsonCapable<AnimalWithTypeIdContainingDot> {
+    /**
+     * Creates an instance of {@link AnimalWithTypeIdContainingDot} by reading the {@link JsonReader}.
+     *
+     * @param jsonReader The {@link JsonReader} that will be read.
+     * @return An instance of {@link AnimalWithTypeIdContainingDot} if the {@link JsonReader} is pointing to
+     * {@link AnimalWithTypeIdContainingDot} JSON content, or null if it is pointing to {@link JsonToken#NULL}.
+     * @throws IllegalStateException If the {@link JsonReader} wasn't pointing to the correct {@link JsonToken} when
+     * passed.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends AnimalWithTypeIdContainingDot> T fromJsonBase(JsonReader jsonReader) {
```

Comment:
Is this intentional (and should it be a pattern) to name it `fromJsonBase`  and not `fromJson`?

Reply:
I didn't like doing this but it was an issue that the subclass also has `fromJson` and you cannot override static methods. This will need to have further design work.

Reply:
Concept has changed and handling is now different